src/Catalyst.Core.Modules.Dfs.Tests/IntegrationTests/CoreApi/BitSwapApiTest.cs

```diff
@@ -0,0 +1,535 @@
+            }
+        }
+
+        [Fact]
+        public async Task GetsBlock_OnRequest()
+        {
+            _dfsService.Options.Discovery.DisableMdns = true;
+            _dfsService.Options.Discovery.BootstrapPeers = new MultiAddress[0];
+            await _dfsService.StartAsync();
+
+            _dfsServiceOther.Options.Discovery.DisableMdns = true;
+            _dfsServiceOther.Options.Discovery.BootstrapPeers = new MultiAddress[0];
+            await _dfsServiceOther.StartAsync();
+            
+            try
+            {
+                var cts = new CancellationTokenSource(10000);
+                var data = Guid.NewGuid().ToByteArray();
+                var cid = await _dfsServiceOther.BlockApi.PutAsync(data, cancel: cts.Token);
+
+                var remote = _dfsServiceOther.LocalPeer;
+                await _dfsService.SwarmApi.ConnectAsync(remote.Addresses.First(), cts.Token);
+
+                var block = await _dfsService.BlockApi.GetAsync(cid, cts.Token);
+                
+                Assert.Equal(cid, block.Id);
+                Assert.Equal(data, block.DataBytes);
+            }
+            finally
+            {
+                await _dfsServiceOther.StopAsync();
+                await _dfsService.StopAsync();
+                _dfsService.Options.Discovery = new DiscoveryOptions();
+                _dfsServiceOther.Options.Discovery = new DiscoveryOptions();
+            }
+        }
+
+        [Fact]
+        public async Task GetsBlock_CidV1()
+        {
+            await _dfsService.StartAsync();
+            await _dfsServiceOther.StartAsync();
+            try
+            {
+                var data = Guid.NewGuid().ToByteArray();
+                var cid = await _dfsServiceOther.BlockApi.PutAsync(data, "raw"); // @TODO get this from a test prop so we can test against multiple hash algos
+
+                var remote = _dfsServiceOther.LocalPeer;
+                await _dfsService.SwarmApi.ConnectAsync(remote.Addresses.First());
+
+                var cts = new CancellationTokenSource(3000);
+                var block = await _dfsService.BlockApi.GetAsync(cid, cts.Token);
+                
+                Assert.Equal(cid, block.Id);
+                Assert.Equal(data, block.DataBytes);
+            }
+            finally
+            {
+                await _dfsServiceOther.StopAsync();
+                await _dfsService.StopAsync();
+            }
+        }
+
+        [Fact]
+        public async Task GetBlock_Timeout()
+        {
+            var block = new DagNode(Encoding.UTF8.GetBytes("BitSwapApiTest unknown block"));
+            await _dfsService.StartAsync();
+            
+            try
+            {
+                var cts = new CancellationTokenSource(300);
+                ExceptionAssert.Throws<TaskCanceledException>(() =>
+                {
+                    var _ = _dfsService.BitSwapApi.GetAsync(block.Id, cts.Token).Result;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable '_'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)